### DISCUSSION TITLE
Delete before deploy and wildcards in delete

### DISCUSSION THREAD

**Discussion by mdicss** [Dynatrace/dynatrace-configuration-as-code#298] [Archive] [answered]
There should be a configuration option to apply a delete configuration **before** the other configurations.
For me it's hard to see the clue to first configure and then delete right afterwards the result in the same deploy statement.
It should be possible to use wildcards in the delete.yaml like:
```
delete:
  - "app-detection-rule/*"
```
to delete all app-detection-rules in this example and then define new app-detection-rules in the same deploy statement.
See my remark on the remaining problem in https://github.com/dynatrace-oss/dynatrace-monitoring-as-code/discussions/293

**Comment by tatomir146** [comment, marked as answer]
Hi @mdicss, for me it looks like you are trying to overcome some configuration or possible api bug here. Monaco is *always* detecting existing configuration by `name` parameter and then using PUT method to update existing, or use POST method to create new configuration. Please make sure to use unmodified `{{ .name }}` variable in JSON and also to define name in YAML file. Having `name` in JSON different then what's defined in YAML would result in this kind of errors you are describing. 

This kind of workflow is also a reason delete is executed after deployment. Otherwise, you would be constantly recreating configuration and for some API's this means losing all history. 

Adding wildcard would IMO make sense. Delete will be reworked with new configuration in monaco 2.0.

**Reply by mdicss** [reply]
Hi @tatomir146 
Actually, I'm testing monaco. So it would be convenient to have the possibility to first delete all 'test'-configurations and then create the new configurations from scratch.
But I see, when losing the history, this maybe an argument, to only delete some specific configurations.

BTW: What about adding the ability to read the configuration files directly from git? Or do you think, this would be bad practice?

**Reply by tatomir146** [reply]
Not sure what exactly you mean by reading configuration directly from git?

**Reply by mdicss** [reply]
For now, I store the yaml and json files for the configuration in git. Then I have to check out the files, see if there are new changes and then run 'monaco deploy'. It could be convenient when monaco could directly get the configuration from git, directly or by check them out then read and delete at the end.

**Reply by tatomir146** [reply]
@mdicss you are missing continuous integration part here. For this you can set up e.g. jenkins job, add git webhook so your job gets triggered every time you e.g. merge configuration to `master` branch. This isn't something monaco should do.

**Reply by tatomir146** [reply]
Delete function discussion is in #218
